myjdapi/exception.py

```diff
@@ -0,0 +1,266 @@
+"""Exceptions of the MyJDownloader API."""
+
+from .const import (
+    EXCEPTION_API_COMMAND_NOT_FOUND,
+    EXCEPTION_API_INTERFACE_NOT_FOUND,
+    EXCEPTION_AUTH_FAILED,
+    EXCEPTION_BAD_PARAMETERS,
+    EXCEPTION_BAD_REQUEST,
+    EXCEPTION_CHALLENGE_FAILED,
+    EXCEPTION_EMAIL_FORBIDDEN,
+    EXCEPTION_EMAIL_INVALID,
+    EXCEPTION_ERROR_EMAIL_NOT_CONFIRMED,
+    EXCEPTION_FAILED,
+    EXCEPTION_FILE_NOT_FOUND,
+    EXCEPTION_INTERNAL_SERVER_ERROR,
+    EXCEPTION_MAINTENANCE,
+    EXCEPTION_METHOD_FORBIDDEN,
+    EXCEPTION_OFFLINE,
+    EXCEPTION_OUTDATED,
+    EXCEPTION_OVERLOAD,
+    EXCEPTION_SESSION,
+    EXCEPTION_STORAGE_ALREADY_EXISTS,
+    EXCEPTION_STORAGE_INVALID_KEY,
+    EXCEPTION_STORAGE_INVALID_STORAGEID,
+    EXCEPTION_STORAGE_KEY_NOT_FOUND,
+    EXCEPTION_STORAGE_LIMIT_REACHED,
+    EXCEPTION_STORAGE_NOT_FOUND,
+    EXCEPTION_TOKEN_INVALID,
+    EXCEPTION_TOO_MANY_REQUESTS,
+    EXCEPTION_UNKNOWN,
+)
+
+
+class MYJDException(BaseException):
+    """Base MyJDownloader Exception."""
+
+    pass
+
+
+class MYJDConnectionException(MYJDException):
+    """Connection Exception."""
+
+    pass
+
+
+class MYJDDeviceNotFoundException(MYJDException):
+    """Device not found Exception."""
+
+    pass
+
+
+class MYJDDecodeException(MYJDException):
+    """Decode Exception."""
+
+    pass
+
+
+class MYJDApiException(MYJDException):
+    """Base MyJDownloader API Exception."""
+
+    @classmethod
+    def get_exception(
+        cls, exception_source, exception_type=EXCEPTION_UNKNOWN, *args, **kwargs
+    ):
+        """Get exception object from MyJDownloader exception type."""
+        return EXCEPTION_CLASSES.get(exception_type.upper(), MYJDUnknownException)(
+            exception_source, *args, **kwargs
+        )
+
+    def __init__(self, exception_source, *args, **kwargs):
+        """Initialize MyJDownloader API exception."""
+        self.source = exception_source.upper()
+        super().__init__(*args, **kwargs)
```

Comment:
If you could change this so it still works with python 2. I know is deprecated/obsolete but since it doesn't take much I prefer to keep compatibility.
The rest seems fine.

 (I think this new github suggestion feature allows you to simply click to modify it, but I have not test it yet)
```suggestion
        super(MYJDApiException, self).__init__(*args, **kwargs)
```